recipes/mpir/all/conanfile.py

```diff
@@ -0,0 +1,96 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, MSBuild
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MpirConan(ConanFile):
+    name = "mpir"
+    description = "MPIR is a highly optimised library for bignum arithmetic" \
+                  "forked from the GMP bignum library."
+    topics = ("conan", "mpir", "multiprecision", "math", "mathematics")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://mpir.org/"
+    license = "LGPL-3.0-or-later"
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    _dll_or_lib = "lib"
+    _source_subfolder = "source_subfolder"
+    _platforms={'x86': 'Win32', 'x86_64': 'x64'}
+
+    def build_requirements(self):
+        self.build_requires("yasm/1.3.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        self._dll_or_lib = "dll" if self.options.shared else "lib"
+
+    @property
+    def _vcxproj_path(self):
+        compiler_version = self.settings.compiler.version if tools.Version(self.settings.compiler.version) < "16" else "15"
+        return os.path.join(self._source_subfolder,"build.vc{}".format(compiler_version),
+                                                   "{}_mpir_gc".format(self._dll_or_lib),
+                                                   "{}_mpir_gc.vcxproj".format(self._dll_or_lib))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], keep_permissions=True)
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _build_visual_studio(self):
+        if "MD" in self.settings.compiler.runtime and not self.options.shared:
+                props_path = os.path.join(self._source_subfolder, "build.vc", 
+                "mpir_{}_{}.props".format(str(self.settings.build_type).lower(), self._dll_or_lib))
+                if self.settings.build_type == "Debug":
+                    tools.replace_in_file(props_path, "<RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>",
```

Comment:
Not completely sure, but may just removing the runtime and letting the MSBuild helper to set it in the command line is enough. Maybe the experienced guys @SSE4 and @uilianries can help here

Reply:
It is not set in the command line, it is set in a temporary .props file that is generated on the fly and passed as an argument (the .props file path). Just in case someone looks for the code.